[docs][joy-ui] Revise the theme typography page (#38285)
mui/material-ui: docs/data/joy/customization/theme-typography/theme-typography.md

<p class="description">Learn about the default theme's typography system and how to customize it.</p>

Joy UI includes a typography system within the theme to help you create consistent text across your application. You can customize the default system or start from scratch depending on your needs.

## Default system

The default system consists of 13 built-in levels:
Joy UI's theme includes, by default, a thoughtfully designed 11 levels typography system to help you create consistent UIs.
Hover over each cell in the table below to see the preview value.

- `body-md` - the baseline typography for the application, used as the default configuration in the [Typography](/joy-ui/react-typography/) and [CssBaseline](/joy-ui/react-css-baseline/) components.
- `body2` through `body5` - can be used for secondary and tertiary information.
- The `h1` to `h6` levels follow the semantic HTML headings.

> - The `display1` and `display2` usually appear as taglines for marketing and landing pages.

Author: Was not feeling like these were adding value ⎯ maybe the "h1 to h6 levels follow the semantic HTML tags" are still useful, but to think about. Open to suggestions!

There are several ways that you can use the theme typography in your application:

#### Typography component

Use the `level` prop in the [Typography](/joy-ui/react-typography/) component:

```jsx
// use the `theme.typography['body-sm']` styles
<Typography level="body-sm">Secondary info</Typography>
```

#### CSS Baseline

The [CSS Baseline](/joy-ui/react-css-baseline/) component applies `body-md` as the default level on the global stylesheet:

```jsx
<CssBaseline />

// inherits the `theme.typography['body-md']` styles
<p>Hello World</p>
```

#### sx prop

> Customize the typographic styles via the `sx` prop using `typography: $level`:

Author: Haha, I think `'some-level'` is indeed clearer ⎯ I'd prefer using that. As a non-developer myself (considering that are designers and others visiting the docs), the `$` symbol is confusing, particularly given it's also used specifically for terminal commands.

```jsx
// to apply the `theme.typography['body-sm']` styles:
<Box sx={{ typography: 'body-sm' }}>Small text</Box>
```

> #### styled function

Reviewer: Here's another crack at it—this one is tough to summarize in a brief headline!

```suggestion
#### Theme typography for custom components
```

Use the [`styled`](/joy-ui/customization/approaches/#reusable-component) function to create a custom component and apply styles from `theme.typography.*`:

```jsx
import { styled } from '@mui/joy/styles';

const Tag = styled('span')((theme) => ({
  ...theme.typography['body-sm'],
  color: 'inherit',
  borderRadius: theme.vars.radius.xs,
  boxShadow: theme.vars.shadow.sm,
  padding: '0.125em 0.375em',
}));
```

## Customizations

> To customize any of the default levels, provide their name as key and an object containing the CSS rules as value to the `theme.typography` node.

Reviewer: ```suggestion
To customize a default level, provide its name as a key along with an object containing the CSS rules as a value to the `theme.typography` node.
```

}
```

### Changing the default font

Joy UI uses the [Inter](https://rsms.me/inter/) font by default.

To change it, override the theme `fontFamily` property:

```js
extendTheme({
  fontFamily: {
    display: 'Noto Sans', // applies to `h1`–`h4`
    body: 'Noto Sans', // applies to `title-*` and `body-*`
  },
});

> ```

Reviewer 2: @danilo-leal Added.

Author: Thank you! 🙏

}
```

### Changing the default font

Joy UI uses the [Inter](https://rsms.me/inter/) font by default.

> To change it, do this...

Author: Here's the specific request left as feedback 😬 @siriwatknp will need your help to quickstart this section!
